src/main/java/ooo/oshi/foreign/windows/Kernel32Library.java

```diff
@@ -60,5 +67,30 @@ public static int getCurrentProcessId() {
 
     private static final MethodHandle getCurrentProcessId = methodHandle("GetCurrentProcessId",
             FunctionDescriptor.of(JAVA_INT));
+    
+    /**
+     * Retrieves the NetBIOS name of the local computer. This name is established at system startup, when the system
+     * reads it from the registry.
+     *
+     * @return the NetBIOS name of the local computer.
+     */
+    public static String getComputerName() {
+    	try {
+        	SegmentAllocator allocator = SegmentAllocator.implicitAllocator();
+    		Addressable buffer = allocator.allocate(2 * (WinBase.MAX_COMPUTERNAME_LENGTH + 1));
+    		Addressable size = allocator.allocate(JAVA_INT, 2 * (WinBase.MAX_COMPUTERNAME_LENGTH + 1));
+            if(!(boolean) getComputerName.invokeExact(buffer, size))	{
```

Comment:
If this fails there will be an error code that you can recover with `getLastError()` (already mapped).  Can you add that to the exception message here?

Reply:
Oops, forgot I left this comment and merged it.... on-call brainfuzz.